src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackTrace.cs

```diff
@@ -225,12 +225,19 @@ internal void ToString(TraceFormat traceFormat, StringBuilder sb)
                     Type? declaringType = mb.DeclaringType;
                     string methodName = mb.Name;
                     bool methodChanged = false;
-                    if (declaringType != null && declaringType.IsDefined(typeof(CompilerGeneratedAttribute), inherit: false))
+
+                    // Removing Debugger support with the ILLinker will remove CompilerGeneratedAttribute.
```

Comment:
I'm a little bit conflicted by this. Up until now, `System.Diagnostics.Debugger.IsSupported` feature basically meant - this is the stuff that is necessary to either attach a debugger at runtime, or to analyze crash dumps post mortem. We have scenarios where removing this support clearly makes sense (WASM, and probably also Xamarin, since (managed) dump debugging where one would be able to use debugger visualizers with Mono is non-existent as far as I know).

But this is about decoding async stack traces that are observable at runtime and are unrelated with the debugging scenarios above. This is the stack traces that people use to implement things like crash telemetry.

Do we really want to lump all of this together? I would also question making this the default now without enough time to get feedback.

Reply:
Unless it is very onerous to add additional checks I'd recommend not lumping this together. Agree with @MichalStrehovsky that stack traces show up in logs, telemetry, Exceptions, and crash reporting, none of which a customer is likely to think about if we say "this scenario doesn't support a debugger."

Reply:
cc @marek-safar who originally proposed removing `CompilerGeneratedAttribute` with https://github.com/dotnet/runtime/pull/39000#discussion_r452362679.

I would be fine with never trimming `CompilerGeneratedAttribute` and `StackTraceHiddenAttribute`. `CompilerGeneratedAttribute` provides a modest size savings in my investigation: roughly `11 KB` uncompressed IL, when compressed the savings are even smaller.